src/ig/IGExporter.ts

```diff
@@ -405,6 +407,76 @@ export class IGExporter {
     });
   }
 
+  /**
+   * Adds any user provided resource files
+   * This includes definitions in:
+   * capabilities, models, operations, profiles, resources, vocabulary, examples
+   *
+   * @param {string} igPath - the path where the IG is exported to
+   */
+  private addPredefinedResources(igPath: string): void {
+    const pathEnds = [
+      'capabilities',
+      'models',
+      'operations',
+      'profiles',
+      'resources',
+      'vocabulary',
+      'examples' // Must come last in case examples are of other resources
+    ];
+    for (const pathEnd of pathEnds) {
+      const dirPath = path.join(this.igDataPath, 'input', pathEnd);
+      if (fs.existsSync(dirPath)) {
+        const files = fs.readdirSync(dirPath);
+        for (const file of files) {
+          let resourceJSON: InstanceDefinition;
+          try {
+            resourceJSON = JSON.parse(fs.readFileSync(path.join(dirPath, file), 'utf-8').trim());
+          } catch (e) {
+            logger.error(`Invalid input file: ${path.join(dirPath, file)}`);
+            continue;
+          }
+          if (resourceJSON.resourceType == null || resourceJSON.id == null) {
+            logger.error(`Resource at ${path.join(dirPath, file)} must define resourceType and id`);
+            continue;
+          }
+          const resource: ImplementationGuideDefinitionResource = {
+            reference: {
+              reference: `${resourceJSON.resourceType}/${resourceJSON.id}`
+            },
+            name: resourceJSON.title ?? resourceJSON.id,
+            description: resourceJSON.description
+          };
+          if (pathEnd === 'examples') {
+            const exampleUrl = resourceJSON.meta?.profile?.find(
+              url =>
+                this.fhirDefs.fishForFHIR(url, Type.Profile) ?? this.pkg.fish(url, Type.Profile)
```

Comment:
This is just a question, but is there a reason to check the `fhirDefs` before the `pkg`? I know there must be, I'm just curious what it is.

Reply:
Ah you have a lot of trust in my reasoning. Maybe too much trust. So my thinking was that it shouldn't matter. Since we are finding based on a URL, a URL should be unique to the profile, so we shouldn't run into naming conflicts.

Reply:
Ah! Good point! Thanks for clarifying!

Reply:
I think the reverse order (`pkg` then `fhirDefs`) might make more sense because there's a pretty good chance it's in the package, and the package is a _lot_ smaller than `fhirDefs`.  Although I _think_ they both use maps for lookup, so I guess maybe size doesn't matter.  I don't know... ?